brigadier/home/views.py

```diff
@@ -1,8 +1,77 @@
 from django.views.generic import TemplateView
+from django.utils import timezone
+from django.db.models import Count, Q
+
+from projects.models import Project, Task, IN_PROGRESS, COMPLETED, NEW
+from employees.models import Employee
 
 
 class HomeView(TemplateView):
     """View of the home page.
 
     """
     template_name = 'home.html'
+    context_object_name = 'statistic_lists'
+
+    def get_context_data(self, **kwargs):
+        """Get context data for home page view:
+            projects_statistics - statistics of the projects
+            tasks_statistics - statistics of the Tasks
+            employees_statistics - statistics of the Employees
+
+        """
+        projects_statistics = Project.objects.aggregate(
+            total=Count('id'),
+            in_progress=Count('id', filter=Q(closed=False)),
+            overdue=Count('id', filter=Q(deadline__lt=timezone.now())),
+            done=Count('id', filter=Q(closed=True)),
+        )
+        tasks_statistics = Task.objects.aggregate(
+            total=Count('id'),
+            in_progress=Count(
+                'id',
+                filter=Q(
+                    status=IN_PROGRESS,
+                    start_date__gte=timezone.now()
+                )
+            ),
+            overdue=Count(
+                'id', filter=Q(
+                    status=IN_PROGRESS,
+                    complete_date__lt=timezone.now()
+                )
+            ),
+            done=Count('id', filter=Q(status=COMPLETED)),
+            new=Count('id', filter=Q(status=NEW)),
+        )
+        employees_statistics = Employee.objects.aggregate(
+            total=Count('id'),
+            occupied=Count(
+                'tasks_assignee',
+                filter=Q(
+                    tasks_assignee__status=IN_PROGRESS,
+                    tasks_assignee__start_date__lte=timezone.now()
+                ),
+                distinct=True
+            ),
+            overdue=Count(
+                'tasks_assignee',
+                filter=Q(
+                    tasks_assignee__status=IN_PROGRESS,
+                    tasks_assignee__complete_date__lte=timezone.now()
+                ),
+                distinct=True
+            ),
+        )
+        employees_total = employees_statistics.get('total')
+        employees_occupied = employees_statistics.get('occupied')
+        employees_statistics['no_tasks'] = (
+            (employees_total if employees_total else 0)
+            - (employees_occupied if employees_occupied else 0)
+        )
```

Comment:
Try to write this code using sql
To negate (invert) the filter use `~` sign in front of Q-expression
I.e.:
```
Count('id', filter=~Q(expression))
```

Reply:
Sometimes it makes sense to do some calculations after the query. I've done it to avoid repeating of complex aggregations in the query. But. aye)

Reply:
Yeah right, but in this case it's better to do so, for reasons:
* it's not a complex computations
* it will not make query running longer (there is only one aggregation in sql, only one group by)
* it'll make your code look cleaner 
* it'll help you to practice using orm

Reply:
I gotcha!